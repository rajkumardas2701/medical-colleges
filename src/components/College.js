import React from 'react';
import PropTypes from 'prop-types';
import '../styles/College.css';

const College = ({ college }) => (
  <a href={`/details/${college.name}`} className="collegeAnc">
    <div className="collegeCard">
      <p className="colgName">
        {college.name}
      </p>
      <p className="colgOwner">
        {college.ownership}
        {' '}
        owned
      </p>
    </div>
  </a>
);

College.propTypes = {
  state: PropTypes.string,
  name: PropTypes.string,
  city: PropTypes.string,
  ownership: PropTypes.string,
  hospitalBeds: PropTypes.string,
}.isRequired;

export default College;
