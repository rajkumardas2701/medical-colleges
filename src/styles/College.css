.collegeCard {
  width: 100%;
  border: 2px solid rgba(41, 29, 29, 0.856);
  border-radius: 5px;
  height: 200px;
  margin: 5px;
  position: relative;
  background-color: rgb(185, 205, 241);
}

.collegeAnc {
  width: 19%;
  text-decoration: none;
}

.colgName {
  position: absolute;
  top: 10px;
  font-size: 25px;
  text-align: center;
  font-family: Arial, Helvetica, sans-serif;
  color: rgba(41, 29, 29, 0.856);
  font-weight: bolder;
}

.colgOwner {
  position: absolute;
  top: 140px;
  left: 30%;
  font-size: 20px;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bolder;
  color: brown;
}