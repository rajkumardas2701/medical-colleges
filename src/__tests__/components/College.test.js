import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import College from '../../components/College';

afterEach(cleanup);

const college = {
  state: 'A & N Islands',
  name: 'Andaman and Nicobar Islands Insitute of Medical Sciences, Port Blair',
  city: 'Port Blair',
  ownership: 'Govt.',
  admissionCapacity: 100,
  hospitalBeds: 460,
};

it('renders item component correctly', () => {
  const drink = renderer.create(<College college={college} />).toJSON();
  expect(drink).toMatchSnapshot();
});

it('renders a link to see college details', () => {
  const { getByTestId } = render(<College college={college} />);
  const link = getByTestId('college-link');
  expect(link.innerHTML).toMatch(college.name);
});
