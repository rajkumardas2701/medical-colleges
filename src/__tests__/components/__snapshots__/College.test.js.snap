// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item component correctly 1`] = `
<a
  className="collegeAnc"
  href="/details/Andaman and Nicobar Islands Insitute of Medical Sciences, Port Blair"
>
  <div
    className="collegeCard"
  >
    <p
      className="colgName"
      data-testid="college-link"
    >
      Andaman and Nicobar Islands Insitute of Medical Sciences, Port Blair
    </p>
    <p
      className="colgOwner"
    >
      Govt.
       
      owned
    </p>
  </div>
</a>
`;
